provider-utils/awscloudformation/utils/video-player-utils.js

```diff
@@ -0,0 +1,126 @@
+  const projectConfigFilePath = context.amplify.pathManager.getProjectConfigFilePath();
+  let projectConfig = fs.readFileSync(projectConfigFilePath, { encoding: 'utf-8' });
+  projectConfig = JSON.parse(projectConfig);
+  return projectConfig;
+}
+
+function fileExtension(framework) {
+  switch (framework) {
+    case 'react':
+      return 'jsx';
+    case 'vue':
+      return 'vue';
+    case 'angular':
+      return 'ts';
+    case 'ember':
+      return 'js';
+    case 'ionic':
+      return 'ts';
+    case 'ios':
+      return 'swift';
+    default:
+  }
+}
+
+function getServiceUrl(amplifyVideoMeta) {
+  const { serviceType, output } = amplifyVideoMeta;
+
+  switch (serviceType) {
+    case 'livestream':
+      return output.oPrimaryMediaStoreEgressUrl;
+    case 'ivs':
+      return output.oVideoOutput;
+    case 'video-on-demand':
+      if (output.oVodOutputUrl) {
+        return `https://${output.oVodOutputUrl}/{path}/{path.m3u8}`;
+      }
+      return output.oVODOutputS3;
+    default:
+  }
+}
+
+function genIosSourcesAndHeaders(context, props, extension) {
+  let template;
+  const { amplify } = context;
+  const projectRootPath = amplify.pathManager.searchProjectRootPath();
+
+  if (extension === 'h') {
+    template = fs.readFileSync(`${__dirname}/../video-player-templates/ios/bridging-header.${extension}.ejs`, { encoding: 'utf-8' });
+    fs.writeFileSync(`${projectRootPath}/${getProjectConfig(context).projectName}/${getProjectConfig(context).projectName}-Bridging-Header.${extension}`, ejs.render(template, props));
+  } else {
+    template = fs.readFileSync(`${__dirname}/../video-player-templates/ios/empty.${extension}.ejs`, { encoding: 'utf-8' });
+    fs.writeFileSync(`${projectRootPath}/${getProjectConfig(context).projectName}/empty.${extension}`, ejs.render(template, props));
+  }
+}
+
+function isVLCKitInstalled(podfile, projectName) {
+  if (podfile.target_definitions[0].children) {
+    const { children } = podfile.target_definitions[0];
+    if (children.length > 0) {
+      return children.some((child) => {
+        if (child.name === projectName) {
+          if (!child.dependencies) {
+            return false;
+          }
+          return child.dependencies.some(dependency => dependency.MobileVLCKit);
+        }
+        return false;
+      });
+    }
+  }
+}
+
+async function installIosDependencies(context) {
+  const { amplify } = context;
+  const projectRootPath = amplify.pathManager.searchProjectRootPath();
+
+  try {
+    fs.readFileSync(`${projectRootPath}/Podfile`, { encoding: 'utf-8' });
+    let podFileData = await exec('pod', ['ipc', 'podfile-json', 'Podfile'], false);
+    podFileData = JSON.parse(podFileData);
+    if (isVLCKitInstalled(podFileData, getProjectConfig(context).projectName)) {
+      context.print.info('Podfile already contains MobileVLCKit');
+      await exec('pod', ['install'], true);
+    } else {
+      const lines = await readLastLines.read(`${projectRootPath}/Podfile`, 1);
```

Comment:
I am worried about this way to edit the Podfile.

My concern is more complicated projects can have potential issues def if they don't follow the typical order:

I.E. They might have multiple targets and the top target is the one we want to add to not the bottom one.

Might be better to use a combo of Regex and prompting the user to ask what target they want to add for the pods file.

React-native does this today with the Podfile: https://github.com/lorenzo-deepcode/react-native/tree/master/local-cli/link/pods

We can follow in their footsteps as a good way to look for the right place to write in the file.

Reply:
Yes you are right, I will refactor this part :)